dipy/reconst/cti.py

```diff
@@ -0,0 +1,658 @@
+        return CorrelationTensorFit(self, params) #ig there's a need to somehow define cti_params here. 
+    
+    # @multi_voxel_fit
+    # def multi_fit(self, data_thres, mask=None):
+
+    #     params = self.fit_method(self.design_matrix, data_thres,
+    #                              self.inverse_design_matrix,
+    #                              weights=self.weights,
+    #                              min_diffusivity=self.min_diffusivity,
+    #                             )
+
+    #     return CorrelationTensorFit(self, params)
+    
+    def predict(self, cti_params, S0=100):  # created
+        """Predict a signal for the CTI model class instance given parameteres
+
+        Parameters:
+        -----------
+        cti_params: numpy.ndarray (...,48)
+                All parameters estimated from the correlation tensor model.
+                Paramters are ordered as follows::
+
+                1. Three diffusion tensor's eigenvalues
+                2. Three lines of the eigenvector matrix each containing the
+                first, second and third coordinates of the eigenvector
+                3. Fifteen elements of the kurtosis tensor
+                4. Twenty-One elements of the covariance tensor
+        gtab1: dipy.core.gradients.GradientTable
+        A GradientTable class instance for first DDE diffusion epoch
+        gtab2: dipy.core.gradients.GradientTable
+        A GradientTable class instance for second DDE diffusion epoch
+        S0 : float or ndarray (optional)
+            The non diffusion-weighted signal in every voxel, or across all
+            voxels. Default: 1
+
+        Returns
+        -------
+        S : numpy.ndarray
+            Signals.
+        """
+
+        return cti_prediction(cti_params, self.gtab1, self.gtab2, S0)
+
+
+class CorrelationTensorFit(DiffusionKurtosisFit):
+
+    """ Class for fitting the Correlation Tensor Model """
+
+    def __init__(self, model, model_params):
+        """ Initialize a CorrelationTensorFit class instance.
+
+        Since CTI is an extension of DKI, class instance is defined as subclass
+        of the DiffusionKurtosis from dki.py
+
+        Parameters
+        ----------
+        model : CorrelationTensorModel Class instance
+            Class instance containing the Correlation Tensor Model for the fit
+        model_params : ndarray (x, y, z, 48) or (n, 48)
+            All parameters estimated from the diffusion kurtosis model.
+            Parameters are ordered as follows:
+                1) Three diffusion tensor's eigenvalues
+                2) Three lines of the eigenvector matrix each containing the
+                   first, second and third coordinates of the eigenvector
+                3) Fifteen elements of the kurtosis tensor
+                4) Twenty One elements of the covariance tensor
+
+        """
+        DiffusionKurtosisFit.__init__(self, model, model_params)
+
+    @property
+    def kt(self):  # self.model_params.shape = (48, )
```

Comment:
I think you don't need to redefine this property here since it is already being imported from the DiffusionKurtosisFit class object. 

Also you have a typo here - the different model parameters should correspond to the last dimension of model_params and not the first. This is being correctly defined in he DiffusionKurtosisFit class object. 

return self.model_params[12:27, ...]  ==> return self.model_params[..., 12:27]